common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -366,23 +366,37 @@ def handle_ajax(self, dispatch, data):
             return json.dumps(self._get_completion(data))
         raise NotFoundError('Unexpected dispatch type')
 
-    def get_metadata(self, view=STUDENT_VIEW):
+    def get_metadata(self, view=STUDENT_VIEW, context=None):
         """Returns a dict of some common block properties"""
-        context = {'exclude_units': True}
+        context = context or {}
+        context['exclude_units'] = True
         prereq_met = True
         prereq_meta_info = {}
         banner_text = None
         display_items = self.get_display_items()
+        course = self._get_course()
+        is_hidden_after_due = False
 
         if self._required_prereq():
             if self.runtime.user_is_staff:
-                banner_text = _('This subsection is unlocked for learners when they meet the prerequisite requirements.')  # lint-amnesty, pylint: disable=line-too-long
+                banner_text = _(
+                    'This subsection is unlocked for learners when they meet the prerequisite requirements.'
+                )
             else:
                 # check if prerequisite has been met
                 prereq_met, prereq_meta_info = self._compute_is_prereq_met(True)
+
+        if prereq_met and view == STUDENT_VIEW and not self._can_user_view_content(course):
+            if context.get('specific_masquerade', False):
+                # Still show the content, but flag to the staff user that the learner wouldn't be able to see it
+                banner_text = self._hidden_content_banner_text(course)
+            else:
+                is_hidden_after_due = True
+
         meta = self._get_render_metadata(context, display_items, prereq_met, prereq_meta_info, banner_text, view)
         meta['display_name'] = self.display_name_with_default
         meta['format'] = getattr(self, 'format', '')
+        meta['is_hidden_after_due'] = is_hidden_after_due
```

Comment:
So this flag is all I'm doing here - letting the MFE know and show an error instead of content.

A _proper_ fix would be also preventing xblocks being rendered in this case, you'd think yeah? Like, it's weird that a user could load the xblock iframe URL and see the content still. However... that's awkward, because this setting (hide-after-due) is a sequence level setting. So an xblock would have to look up the hierarchy and do something based on that. Which isn't insane... but it's work. And the legacy experience has this same issue.

So I just went for parity with legacy here.